### ISSUE TITLE
Dynamically loading GWT assets

### ISSUE THREAD

**Issue by EoinF** [libgdx/libgdx#5253]
### The Problem
Currently the way things are set up in the gwt-backend, all files are loaded upon starting the application. Once this is complete, no more files can be loaded due to the way GwtFileHandle is implemented.
The files available to load through the FileHandle constructor are tied to `com.badlogic.gdx.backends.gwt.preloader.Preloader` class which currently has no api for loading extra files once it has completed the initial load.

### Potential Solutions
- Expose a `loadAsset` method on the Preloader class which takes a link String and a callback function as arguments
- Add an overload to the `preload` method with an extra argument `String[] extraAssetLinks`

Personally, I would prefer the former, although it would require some refactoring to keep the `Preloader ` class DRY, the place where each asset is loaded individually could be extracted into its own function for example.

I am willing make this change myself, but I was wondering if it was more sensible to simply create a subclass of `Preloader` in my own project instead.

**Comment by MobiDevelop** [comment]
There is a way to make different bundles for the preloader to defer loading of assets until a time that is more appropriate. Most of should still apply.

There is currently an error on my blog but the way back machine looks to have a copy: 
http://web.archive.org/web/20171021204851/http://nexsoftware.net:80/wp/2013/07/07/libgdx-deferred-asset-loading-with-gwt/

**Comment by MrStahlfelge** [comment]
The issue has been adressed on different other issues and PRs.
* If you need the feature to not load everything at startup, but have a closed set of assets to load dynamically, PR #5677 is for you - please review and vote to get it merged
* If you need real dynamics to fetch images not known at compile time, issue #6023 is for you

Closing this as a dublicate.
